audioipc/src/shm.rs

```diff
@@ -76,6 +75,19 @@ mod unix {
             }
         }
 
+        let id_cstring = std::ffi::CString::new(id).unwrap();
```

Comment:
nit: It seems we can move this `let id_cstring = std::ffi::CString::new(id).unwrap()` to the beginning of this function and remove the `let id_cstring = std::ffi::CString::new(id).unwrap()` in `#[cfg(target_os = "linux")]` block.